File: rcc/communications/client/dbus/asus/asus_base_client.py
from rcc.communications.client.dbus.abstract_dbus_client import AbstractDbusClient


class AsusBaseClient(AbstractDbusClient):
    """DBus platform client"""

    def __init__(
        self, interface_sufix: str, object_path_sufix: str = "", required=True
    ):  # pylint: disable=too-many-arguments,too-many-positional-arguments
        object_sufix = object_path_sufix
        if object_sufix != "":
            object_sufix = "/" + object_sufix

        super().__init__(
            True,
            "xyz.ljones.Asusd",
            "/xyz/ljones" + object_sufix,
            "xyz.ljones." + interface_sufix,
            required,
        )

File: RogControlCenter/src/clients/dbus/asus/asus_base_client.cpp
#include "../../../../include/clients/dbus/asus/asus_base_client.hpp"

AsusBaseClient::AsusBaseClient(std::string interface_sufix, std::string object_path_sufix, bool required)
	: AbstractDbusClient(true, QString("xyz.ljones.Asusd"),
						 QString::fromStdString("/xyz/ljones" + (object_path_sufix.empty() ? "" : "/" + object_path_sufix)),
						 QString::fromStdString("xyz.ljones." + interface_sufix), required) {
}